portfolio/src/main/webapp/script.js

```diff
@@ -60,6 +51,61 @@ async function getCommentFromFetch() {
         var para = document.createElement("p");
         var node = document.createTextNode(commentString);
         para.appendChild(node);
+        var button = document.createElement("button");
+        button.innerHTML = "Like";
+        const input = commentString.slice();
+        button.addEventListener("click", function () { likeComments(input) });
         commentContainer.appendChild(para);
+        commentContainer.appendChild(button);
     };
+}
+async function getCommentFromFetch() {
+    console.log("Try fetch");
+    const response = await fetch("/data");
+    const comments = await response.json();
+    const dummyElement = comments.pop();
+    var maxComments = parseInt(dummyElement.timestamp, 10);
+    if (Number.isNaN(maxComments)) {
+        console.log("You are trying to input a non-numeric data, please try again");
+        // Set the limit to default value
+        maxComments = 10;
+    }
+    const commentLimitContainer = document.getElementById("comment-limit");
+    commentLimitContainer.value = (maxComments.toString());
+    fetchHelper(maxComments, comments);
+}
+
+async function onChangeFromFetch() {
```

Comment:
Yeah the function name is not that great. I will change that but the two functions cannot be combined because the way they take in the number of maximum comments is different. They both need to display all comments so I write a helper for that. But one function takes in the maximum count stored in the datastore while the other function takes in the maximum count given in the input box. These two functions support two distinct features of my website. Thus, the two functions cannot be combined. I can only reduce some code duplication with the helper function.